File: sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    int count=0;
    public int SumNumbers(TreeNode root) {
        preorder(root,0);
        return count;
        
    }
    
    public void preorder(TreeNode r, int currNumber)
    {
        if(r!=null)
        {
            currNumber = currNumber *10 + r.val;
            if(r.left==null && r.right==null)
            {
                count+=currNumber;
            }
            preorder(r.left,currNumber);
            preorder(r.right,currNumber);
        }
    }
}

File: assignments/src/Sum_root_to_leaf_numbers.java
public class Sum_root_to_leaf_numbers {

    // Time: O(n), Space: O(h) where h = height of the tree
    int totSum = 0;
    public int sumNumbers(TreeNode root) {

        travel(root, 0);
        return totSum;
    }

    void travel(TreeNode root, int currNum) {
        if (root == null) return;

        currNum = currNum*10 + root.val;

        if (root.left == null && root.right == null) {
            totSum += currNum;
        }

        travel(root.left, currNum);
        travel(root.right, currNum);
    }
}
